### PR TITLE
Flatpak & CI improvements

### PR DESCRIPTION
**Pull request by GeorgesStavracas**
Generate Flatpak bundles for all manifests we have in-tree.

Cleanup and increment the builds to include:

 * Ubuntu 20.04
 * Ubuntu 21.10
 * Fedora 35

### CODE REVIEW

**Comment by GeorgesStavracas** [comment]
I'm not ecstatic about fighting Ubuntu 18.04 to check for the different backends, but at least all backends are compiled by the Flatpak jobs.

**Comment by TingPing** [comment]
> I'm not ecstatic about fighting Ubuntu 18.04 to check for the different backends

I'm totally ok with adding newer Ubuntu, on top of that one, there.

```yaml
runs-on: ubuntu-latest
container: ubuntu:21.04
```

For example. (Note the apt commands need to be tweaked a bit, see https://github.com/flatpak/xdg-desktop-portal/blob/master/.github/workflows/check.yml)

File: .github/workflows/flatpak.yml [hunk 1]
```diff
@@ -0,0 +1,63 @@
+---
+
+name: Flatpak
```
Comment (TingPing) [inline .github/workflows/flatpak.yml:3]: Name this "Portal Test Flatpak", more clear what it is.

Comment (GeorgesStavracas) [reply .github/workflows/flatpak.yml]: Sure

**Comment by TingPing** [comment]
Note that 21.04 and 21.10 have GTK4 while 20.04 doesn't.

**Comment by GeorgesStavracas** [comment]
This is roughly the limit I'm comfortable growing the `build.yml` file. If we want more distros, it might be better to switch to an OS matrix or any other fancy technique.

**Review by TingPing — APPROVED** [review]
Looks good.

**Comment by TingPing** [comment]
Whats the plan for building the qt backend?

**Comment by GeorgesStavracas** [comment]
> Whats the plan for building the qt backend?

I'm not touching anything Qt-related anymore until @grulja lands the improved `libportal-qt5` with more API coverage, etc.

**Pull request merged**